agent/api/appmesh/appmesh_test.go

```diff
@@ -0,0 +1,173 @@
+	assert.Equal(t, mockIgnoredGID, appMesh.IgnoredGID)
+	assert.Equal(t, mockProxyEgressPort, appMesh.ProxyEgressPort)
+	assert.Equal(t, mockProxyIngressPort, appMesh.ProxyIngressPort)
+	assert.Equal(t, mockAppPort1, appMesh.AppPorts[0])
+	assert.Equal(t, mockAppPort2, appMesh.AppPorts[1])
+	assert.Equal(t, mockEgressIgnoredIP1, appMesh.EgressIgnoredIPs[0])
+	assert.Equal(t, mockEgressIgnoredIP2, appMesh.EgressIgnoredIPs[1])
+	assert.Equal(t, mockEgressIgnoredPort1, appMesh.EgressIgnoredPorts[0])
+	assert.Equal(t, mockEgressIgnoredPort2, appMesh.EgressIgnoredPorts[1])
+}
+
+func TestAppMeshFromACSNoEgressIgnoredPorts(t *testing.T) {
+	testProxyConfig := prepareProxyConfig()
+	delete(testProxyConfig.Properties, egressIgnoredPorts)
+
+	appMesh, err := AppMeshFromACS(&testProxyConfig)
+
+	assert.NoError(t, err)
+	assert.NotNil(t, appMesh)
+	assert.True(t, appMesh.AppMeshEnabled)
+	assert.Equal(t, mockIgnoredUID, appMesh.IgnoredUID)
+	assert.Equal(t, mockIgnoredGID, appMesh.IgnoredGID)
+	assert.Equal(t, mockProxyEgressPort, appMesh.ProxyEgressPort)
+	assert.Equal(t, mockProxyIngressPort, appMesh.ProxyIngressPort)
+	assert.Equal(t, mockAppPort1, appMesh.AppPorts[0])
+	assert.Equal(t, mockAppPort2, appMesh.AppPorts[1])
+	assert.Equal(t, mockEgressIgnoredIP1, appMesh.EgressIgnoredIPs[0])
+	assert.Equal(t, mockEgressIgnoredIP2, appMesh.EgressIgnoredIPs[1])
+	assert.Equal(t, 0, len(appMesh.EgressIgnoredPorts))
+}
+
+func TestAppMeshFromACSNoEgressIgnoredIPs(t *testing.T) {
+	testProxyConfig := prepareProxyConfig()
+	delete(testProxyConfig.Properties, egressIgnoredIPs)
+
+	appMesh, err := AppMeshFromACS(&testProxyConfig)
+
+	assert.NoError(t, err)
+	assert.NotNil(t, appMesh)
+	assert.True(t, appMesh.AppMeshEnabled)
+	assert.Equal(t, mockIgnoredUID, appMesh.IgnoredUID)
+	assert.Equal(t, mockIgnoredGID, appMesh.IgnoredGID)
+	assert.Equal(t, mockProxyEgressPort, appMesh.ProxyEgressPort)
+	assert.Equal(t, mockProxyIngressPort, appMesh.ProxyIngressPort)
+	assert.Equal(t, mockAppPort1, appMesh.AppPorts[0])
+	assert.Equal(t, mockAppPort2, appMesh.AppPorts[1])
+	assert.Equal(t, 0, len(appMesh.EgressIgnoredIPs))
+	assert.Equal(t, mockEgressIgnoredPort1, appMesh.EgressIgnoredPorts[0])
+	assert.Equal(t, mockEgressIgnoredPort2, appMesh.EgressIgnoredPorts[1])
+}
+
+func TestAppMeshFromACSNonAppMeshProxyInput(t *testing.T) {
+	someOtherProxyType := "fooProxy"
+	testProxyConfig := prepareProxyConfig()
+	testProxyConfig.Type = &someOtherProxyType
+
+	appMesh, err := AppMeshFromACS(&testProxyConfig)
+
+	assert.NoError(t, err)
+	assert.False(t, appMesh.AppMeshEnabled)
+}
+
+func TestAppMeshFromACSNoUIDGID(t *testing.T) {
+	testProxyConfig := prepareProxyConfig()
+
+	delete(testProxyConfig.Properties, ignoredGID)
+	delete(testProxyConfig.Properties, ignoredUID)
+
+	appMesh, err := AppMeshFromACS(&testProxyConfig)
+
+	assert.Error(t, err)
+	assert.Nil(t, appMesh)
+}
+
+func TestAppMeshFromACSNpProxyIngressPort(t *testing.T) {
```

Comment:
**TestAppMeshFromACSNpProxyIngressPort** should be **TestAppMeshFromACSNoProxyIngressPort**?